src/components/forms/DataLocation.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {useCallback} from 'react';
+import {FormField, FormFieldTypes, FormValidators} from './forms';
+import {cloneDeep} from 'lodash/fp';
```

Comment:
Consider using `lodash` instead of `lodash/fp`, `lodash/fp` isn't really supported anymore and sometimes doesn't behave well with types.